include/Transport.hpp

```diff
@@ -170,6 +213,8 @@ namespace mediasoupclient
 		  const std::string& kind,
 		  nlohmann::json* rtpParameters,
 		  const nlohmann::json& appData = nlohmann::json::object());
+		
+		DataConsumer* ConsumeData(DataConsumer::Listener* listener,	DataConsumerOptions dataConsumerOptions);
```

Comment:
In libmediasoupclient we don't use `ProducerOptions` `ConsumerOptions` but the constructor takes the needed arguments separately: dataConsumerId, dataProducerId, sctpStreamParameters, label, etc.

Lets build the `DataChannelInit` instance internally on constructor with the given arguments. There is no need to expose it to the user in this case.

Reply:
OK, done. But I guess you mean the ProduceData()/ConsumeData() take the needed arguments and not the constructors, right?